packages/mds-agency/handlers/create-event.ts

```diff
@@ -138,20 +138,28 @@ export const createEventHandler = async (
     await refreshDeviceCache(device, event)
 
     const { telemetry } = event
+    let recorded_telemetry: Recorded<Telemetry> | undefined = undefined
     if (telemetry) {
-      await db.writeTelemetry(normalizeToArray(telemetry))
+      const results = await db.writeTelemetry(normalizeToArray(telemetry))
+      if (results.length > 0) {
+        recorded_telemetry = results[0]
+      }
     }
```

Comment:
Might be a bit nicer to make this a closure (github won't let me do a suggestion), just to avoid potential mutation of `recorded_telemetry`.

Something like:
```typescript
const [recordedTelemetry] = telemetry ? await db.writeTelemetry(normalizeToArray(telemetry)) : []
// recordedTelemetry will infer a type of Recorded<Telemetry> | undefined 
```